Applet/src/com/android/javacard/keymaster/KMKeymasterApplet.java

```diff
@@ -3876,31 +3947,35 @@ private static void encryptSecret(byte[] scratchPad) {
     data[SECRET] = KMByteBlob.instance(scratchPad, (short)0, len);
   }
 
-  private static void makeAuthData(byte[] scratchPad) {
-    short arrayLen = 2;
-    if (KMArray.cast(data[KEY_BLOB]).length() == 5) {
-      arrayLen = 3;
+  private static void makeAuthData(short version) {
```

Comment:
take a input parameter for keyType here.

Reply:
optmized the code.